src/server/api/hold.ts

```diff
@@ -230,3 +176,65 @@ export async function postEDDRequest(
     }
   }
 }
+
+/**
+ * Getter function for hold request details.
+ */
+// TODO: Add return type
+export async function fetchHoldDetails(
+  requestId: string
+): Promise<HoldDetailsResult> {
+  try {
+    const client = await nyplApiClient()
+    const holdDetailsResult = await client.get(`/hold-requests/${requestId}`)
+    const { id, pickupLocation, patron } = holdDetailsResult.data
+
+    return {
+      requestId: id,
+      patronId: patron,
+      pickupLocation,
+      status: 200,
+    }
+  } catch (error) {
+    console.error(
+      `Error fetching hold request details in fetchHoldRequestDetails server function, requestId: ${requestId}`,
+      error.message
+    )
+
+    return { status: 500 }
+  }
+}
+
+/**
+ * Getter function for hold request eligibility for patrons.
+ */
+export async function fetchPatronEligibility(
+  patronId: string
+): Promise<PatronEligibilityStatus> {
+  const eligibilityEndpoint = `/patrons/${patronId}/hold-request-eligibility`
+
+  try {
+    const client = await nyplApiClient()
+    const eligibilityResult = await client.get(eligibilityEndpoint, {
+      cache: false,
+    })
+
+    // There should always be en eligibilty boolean attribute returned from Discovery API
+    if (eligibilityResult.eligibility === undefined) {
+      throw new Error("Improperly formatted eligibility from Discovery API")
+    }
+
+    if (eligibilityResult.eligibility === false) {
+      return { status: 401, ...eligibilityResult }
+    }
+
+    return { status: 200, ...eligibilityResult }
+  } catch (error) {
+    console.error(
+      `Error fetching hold request eligibility in fetchPatronEligibility server function, patronId: ${patronId}`,
+      error.message
+    )
+
+    return { status: 500 }
```

Comment:
Can you add a message and say it was a server issue? It seems this would be the case when the NYPL Platform eligibility endpoint is down, right?